tests/flow/test_optimizations_plan.py

```diff
@@ -354,3 +354,37 @@ def test22_no_expand_into_across_scopes(self):
                     [14, 0, 2],
                     [14, 0, 3]]
         self.env.assertEqual(resultset, expected)
+
+    # Constructions of the form:
+    # MATCH (a) WHERE a:A
+    # should utilize a Label Scan rather than an All Node Scan.
+    def test23_has_label_to_label_scan(self):
+        # Verify that the optimization is applied properly.
+        query = """MATCH (a) WHERE a:person RETURN a.val ORDER BY a.val LIMIT 3"""
+        executionPlan = graph.execution_plan(query)
+        self.env.assertNotIn("All Node Scan", executionPlan)
+        self.env.assertIn("Label Scan", executionPlan)
+        resultset = graph.query(query).result_set
+        expected = [[0],
+                    [1],
+                    [2]]
+        self.env.assertEqual(resultset, expected)
+
+        # The optimization should apply when one of multiple AND-related filters is a label specification.
+        query = """MATCH (a) WHERE a:person AND a.val = 0 RETURN a.val ORDER BY a.val LIMIT 3"""
+        executionPlan = graph.execution_plan(query)
+        self.env.assertNotIn("All Node Scan", executionPlan)
+        self.env.assertIn("Label Scan", executionPlan)
+        resultset = graph.query(query).result_set
+        expected = [[0]]
+        self.env.assertEqual(resultset, expected)
+
+        # The optimization is not valid for OR-specified multiple labels.
+        query = """MATCH (a) WHERE a:person or a:FakeLabel RETURN a.val ORDER BY a.val LIMIT 3"""
+        executionPlan = graph.execution_plan(query)
+        self.env.assertIn("All Node Scan", executionPlan)
+        resultset = graph.query(query).result_set
+        expected = [[0],
+                    [1],
+                    [2]]
+        self.env.assertEqual(resultset, expected)
```

Comment:
I think we should have tests covering:
1. `MATCH (n:L) WHERE n:K RETURN *`
2. `MATCH (n) WHERE n:K AND n:L RETURN *`
2. `MATCH (n:L {id:1})-[]->(m) WHERE m:K RETURN *`